src/random_test.cpp

```diff
@@ -2679,11 +2679,16 @@ void Table::Compare_between_engine(const std::string &sql, Thd1 *thd) {
   set_default();
 }
 
+  // Data structures for recent queries
+  static std::deque<std::string> recent_queries;
+
 bool execute_sql(const std::string &sql, Thd1 *thd) {
   auto query = sql.c_str();
   static auto log_all = opt_bool(LOG_ALL_QUERIES);
   static auto log_failed = opt_bool(LOG_FAILED_QUERIES);
   static auto log_success = opt_bool(LOG_SUCCEDED_QUERIES);
+  static auto log_N = opt_bool(LOG_N_QUERIES);
+  thd->max_recent_queries = options->at(Option::LOG_N_QUERIES)->getInt();
```

Comment:
you are reassigning this variable . This is done as part of Thd class initialization as well

Reply:
Yes, removed